src/main/java/io/vertx/circuitbreaker/impl/CircuitBreakerImpl.java

```diff
@@ -228,6 +233,23 @@ private synchronized CircuitBreaker attemptReset() {
     return this;
   }
 
+  private <T> Future<T> retryFuture(AtomicInteger retryCount, Handler<Future<T>> operation, Future<T> operationResult) {
```

Comment:
rather use an `int` than a mutable `AtomicInteger` for `retryCount`, as you pass `retryFuture` is called anyway with an incremented retry count value